### PR TITLE
resource: add permission check to names search

### PR DESCRIPTION
**Pull request by jrcastro2**
* closes https://github.com/CERNDocumentServer/cds-rdm/issues/191

### CODE REVIEW

**Review by alejandromumo — COMMENTED** [review]
LGTM, I am pinging Alex for a double check that the endpoint can be also protected by login on Zenodo

WDYT @slint

File: invenio_vocabularies/contrib/names/resources.py [hunk 1]
```diff
@@ -52,3 +61,40 @@ def name_resolve_by_id(self):
         )
 
         return item.to_dict(), 200
+
+    @request_extra_args
+    @request_search_args
+    @response_handler(many=True)
+    @login_required
```
Comment (slint) [inline invenio_vocabularies/contrib/names/resources.py:68]: minor: shouldn't we instead be using the permission policy and the `AuthenticatedUser` need? If this is how this is done in other places, feel free to ignore :)

**Comment by slint** [comment]
As discussed IRL, the change is fine, especially for the `names` vocabulary since it's only used in forms that require authentication anyways.

File: invenio_vocabularies/contrib/names/resources.py (hunk 1 above)
Comment (jrcastro2) [reply invenio_vocabularies/contrib/names/resources.py:68]: That's a good point! 
It really depends on whether we want to protect just the endpoint, as the login_required decorator does, or if we also want to secure the service methods. I believe we should go with the latter approach, so I’ll make the necessary changes based on your suggestion. Thanks!

**Review by slint — APPROVED** [review]

**Pull request merged**